src/controllers/auth.js

```diff
@@ -0,0 +1,77 @@
+const jwt = require("jsonwebtoken");
+const bcrypt = require("bcrypt");
+const fileWriter = require("../helper/file-helper");
+
+const validator = require("../helper/validator");
+//Tasks Data
+const usersData = require("../data/users.json");
+
+const registerUser = (req, res) => {
+  try {
+    let userDetails = req.body;
+    let { users } = usersData;
+    let validatorObj = validator.validateUserInfo(userDetails, users);
+    if (validatorObj.status) {
+      userDetails.password = bcrypt.hashSync(userDetails.password, 8);
+      let result = fileWriter(userDetails, users);
+      if (result.status) {
+        return res.status(200).send(result);
+      } else {
+        return res.status(500).send(result);
+      }
+    } else {
+      res.status(400);
+      return res.send(validatorObj);
+    }
+  } catch (error) {
+    return res.status(500).send({ message: "Something went wrong !" });
+  }
+};
+
+const loginUser = (req, res) => {
+  try {
+    let currUser = req.body;
+    let { users } = usersData;
+
+    let dbUser = users.find((item) => item.email == currUser.email);
+    if (dbUser) {
```

Comment:
try to avoid long if else blocks. A better structure would have been something as follows -:

`if(!dbUser) return res.status(404).send({ message: "User Not Found" });` and all the other logic would have been outside the if block.